sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ApplicationGatewayRequestRoutingRule.java

```diff
@@ -389,6 +394,27 @@ interface DefinitionStages {
              */
             WithAttach<ParentT> withUrlPathMap(String urlPathMapName);
         }
+
+        /**
+         * The stage of an application gateway request routing rule definition allowing to associate the rule with a
+         * priority.
+         * @param <ParentT> the stage of the application gateway definition to return to after attaching this definition
+         */
+        interface WithPriority<ParentT> {
+            /**
+             * ({@link ApplicationGatewaySkuName#STANDARD_V2} and {@link ApplicationGatewaySkuName#WAF_V2} Only)
+             * Specifies a unique priority value for the request routing rule.
+             * <p>If you don't assign the priority, SDK will auto-assign a unique value for you (ranging from 10010 to
+             * 20000) in the ordering of definition. For rules with auto-assigned priorities, those which defined later
+             * will have larger priority values (lower priority) over those which defined earlier.</p>
+             * <p>If you want to assign it manually, unless you want the rule to have a lower priority over the others,
+             * we recommend a value no larger than 10000 to avoid potential priority collision with rules with
+             * auto-assigned priorities.</p>
```

Comment:
Maybe just say that user should always specify the priority (if applicable).

After that, mention SDK's default value (but we want it happen as little as possible.